diff --git a/platform/clounix/clounix-modules/modules/src/clx_dev/osal_mdc.c b/platform/clounix/clounix-modules/modules/src/clx_dev/osal_mdc.c
index 008eb60f8..3872e255f 100644
--- a/platform/clounix/clounix-modules/modules/src/clx_dev/osal_mdc.c
+++ b/platform/clounix/clounix-modules/modules/src/clx_dev/osal_mdc.c
@@ -794,8 +794,11 @@ _osal_mdc_maskStatus(
     {
         /* Mask */
         pci_read_config_dword(ptr_rc_dev, ext_cap + 0x8, &data_32);
-        data_32 |= 0x20;
+        data_32 |= 0x4021;
         pci_write_config_dword(ptr_rc_dev, ext_cap + 0x8, data_32);
+        pci_read_config_dword(ptr_rc_dev, ext_cap + 0x14, &data_32);
+        data_32 |= 0x1;
+        pci_write_config_dword(ptr_rc_dev, ext_cap + 0x14, data_32);
     }
 
     return CLX_E_OK;
@@ -816,12 +819,16 @@ _osal_mdc_clearStatus(
         /* Clear */
         pci_write_config_word(ptr_rc_dev, ptr_rc_dev->pcie_cap + 0xa, 0x04);
         pci_write_config_word(ptr_rc_dev, ptr_rc_dev->pcie_cap + 0x12, 0x8000);
-        pci_write_config_dword(ptr_rc_dev, ext_cap + 0x4, 0x20);
+        pci_write_config_dword(ptr_rc_dev, ext_cap + 0x4, 0x4021);
+        pci_write_config_dword(ptr_rc_dev, ext_cap + 0x10, 0x1);
 
         /* UnMask */
         pci_read_config_dword(ptr_rc_dev, ext_cap + 0x8, &data_32);
         data_32 &= ~0x20;
         pci_write_config_dword(ptr_rc_dev, ext_cap + 0x8, data_32);
+        pci_read_config_dword(ptr_rc_dev, ext_cap + 0x14, &data_32);
+        data_32 &= ~0x1;
+        pci_write_config_dword(ptr_rc_dev, ext_cap + 0x14, data_32);
     }
 
     return CLX_E_OK;
